src/geocat/comp/__init__.py

```
@@ -3,7 +3,115 @@
+
+            Note:
+                If left unspecified, the rightmost coordinate dimension of fi
+                will be used. This parameter must be specified as a keyword
+                argument.
+
+        yi (:class:`numpy.ndarray`):
+            An array that specifies the Y coordinates of the fi array. Most
+            frequently, this is a 1D strictly monotonically increasing array
+            that may be unequally spaced. In some cases, yi can be a multi-
+            dimensional array (see next paragraph). The rightmost dimension
+            (call it nyi) must have at least two elements, and is the second-
+            to-last dimension of fi.
+
+            If yi is a multi-dimensional array, then each nyi subsection of yi
+            must be strictly monotonically increasing, but may be unequally
+            spaced. All but its rightmost dimension must be the same size as
+            all but fi's rightmost two dimensions.
+
+            For geo-referenced data, yi is generally the latitude array.
+
+            Note:
+                If left unspecified, the second-to-rightmost coordinate
+                dimension of fi will be used. This parameter must be specified
+                as a keyword argument.
+
+    Returns:
+
+        :class:`xarray.DataArray`: The interpolated grid. If the *meta*
+        parameter is True, then the result will include named dimensions
+        matching the input array. The returned value will have the same
+        dimensions as fi, except for the rightmost two dimensions which
+        will have the same dimension sizes as the lengths of yo and xo.
+        The return type will be double if fi is double, and float otherwise. 
+
+    """
+
+    if xmsg is None:
+        xmsg = _ncomp.dtype_default_fill[fi.dtype]
+
+    if xi is None:
+        xi = fi.coords[fi.dims[-1]].values
+    if yi is None:
+        yi = fi.coords[fi.dims[-2]].values
+    fi_data = fi.data
+    fo_chunks = list(fi.chunks)
```

Per our discussion the other day my understanding is that because the  input interpolation dimensions (two fastest varying dimensions) do not match the output interpolation dimensions the data cannot be blocked along the interpolation dimensions. What will happen if the user provides an input array that *is* blocked along the interpolation dimensions ?

---

In commit 28c5d94 I added a check to ensure that the chunked shapes of the interpolation (fastest varying) dimensions of the input array always match the full shape of those dimensions. Existing chunking across any additional dimensions will remain intact.

---

Is it reasonable (good behavior) to re-chunk the users input array? This seems like an unintended side effect of using linint. Maybe we should just throw an error...

---

Re-chunking should not actually modify the chunking on the original array, but it *may* create a duplicate of the input array temporarily if the newly defined chunks are not already contiguous in memory.

Regardless, you make a good point that it might be better behavior to just throw an error. As of commit 30d85f5, I've defined a new `ChunkError` exception to be thrown when an input array in chunked in a way that is incompatible with an underlying Fortran routine, and I removed the re-chunking step and now throw a `ChunkError` instead.